From 14059d05ac2a444ab7eee781cb05aa006c259094 Mon Sep 17 00:00:00 2001
From: Franck Jullien <franck.jullien@odyssee-systemes.fr>
Date: Sun, 17 May 2020 21:46:59 -0700
Subject: [PATCH 1365/1852] v4l: xilinx: tpg: Don't read version if HLS IP

commit 2e2d1eeb7e2efb660a968c65c5fecb8934c5d31e from
https://github.com/Xilinx/linux-xlnx.git

Version register is no longer present in HLS version of TPG.

Signed-off-by: Franck Jullien <franck.jullien@odyssee-systemes.fr>
Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index f840bc098d9e..0beb153d4555 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -1225,7 +1225,8 @@ static int xtpg_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, xtpg);
 
-	xvip_print_version(&xtpg->xvip);
+	if (!xtpg->is_hls)
+		xvip_print_version(&xtpg->xvip);
 
 	/* Initialize default frame interval */
 	xtpg->fi_n = 1;
-- 
2.31.1

